Wrap file opening errors so that proper HTTP status codes can be given when using http.FS

jasonhancock opened the pull request:
I had a problem when using a LayerFs with http.FS to serve up assets. If someone requested an asset that did not exist, instead of getting a 404 they would get a 500. By wrapping the errors, the http.FS code can now Unwrap() them to get the underlying fs error and determine the proper status code to serve.

jasonhancock committed:
Wrap file opening errors so that proper HTTP stats codes can be given when using http.FS

dschmidt commented:
This looks great - unfortunately the CI fails... maybe unwrapping the error only happens in a newer go version? 

Feel free to bump the version requirement if necessary

jasonhancock commented:
It looks like the necessary change isn't related to Unwrap (go 1.13) but was incorporated in net/http in  go 1.17. As go.mod already sets the Go version to 1.17, I bumped the version in the workflow file to 1.17 to match. Can you approve/kick off the CI jobs again?

dschmidt commented:
lint and coverage are now/still unhappy, not at my desk - but I can probably approve ci runs a few more times 😅

or have a look myself tomorrow

jasonhancock committed:
Update actons workflows

Bump versions of checkout and go-setup to get off node 16 based actions.
Update coverage to use `go install` instead of `go get`

jasonhancock commented:
Pushed a commit with what I think are fixes for the workflow files.

dschmidt commented:
still broken unfortunately, I can check tomorrow

jasonhancock commented:
Think I might have it... forgot the `@latest` tag on the `go install`, and removed `ifshort` from the golangci config.

dschmidt commented:
Great! Thanks a lot - in addition to the neat contribution I honestly appreciate the effort to get the ci up and running again 🤗

dschmidt commented:
In case you need it, I released [v0.2.0](https://github.com/dschmidt/go-layerfs/releases/tag/v0.2.0)
